File: latte/src/main/java/gg/beemo/latte/broker/Exceptions.kt
package gg.beemo.latte.broker

import gg.beemo.latte.broker.rpc.RpcStatus

sealed class BrokerException(message: String?) : Exception(message)
class RpcRequestTimeout(message: String) : BrokerException(message)
class IgnoreRpcRequest : BrokerException("Ignoring RPC request")
class RpcException(val status: RpcStatus) : BrokerException("RPC failed with status $status")

File: water/src/broker/Exceptions.ts
import type { RpcStatus } from "../index.js";

export class BrokerException extends Error {
    public constructor(message?: string) {
        super(message);
    }
}

export class RpcRequestTimeout extends BrokerException {
    public constructor(message: string) {
        super(message);
    }
}

export class IgnoreRpcRequest extends BrokerException {
    public constructor() {
        super("Ignoring RPC request");
    }
}

export class RpcException extends BrokerException {
    public constructor(public readonly status: RpcStatus) {
        super(`RPC failed with status ${status}`);
    }
}
